internal/configs/provider.go

```diff
@@ -290,3 +325,79 @@ func checkProviderNameNormalized(name string, declrange hcl.Range) hcl.Diagnosti
 	}
 	return diags
 }
+
+func (p *ProviderBlock) generateForEachProviders(eval *StaticEvaluator) ([]*Provider, hcl.Diagnostics) {
```

Comment:
Can we keep the methods of the `ProviderBlock` struct close to each other and to the struct itself?
Or else it's a bit confusing when there are similar structs in the same file with mixed functions between them (also relevant for `providerBlockSchema`, which is only relevant to the `ProviderBlock`)

Reply:
ok, new layout has the following reasons: 

- Schema at the bottom due to convention. 
- ProviderBlock functions above that to keep them grouped
- Provider functions above that to keep struct methods together
- Private functions above that
- Public functions above that. because that felt right, but is probably some java convention that I remember from the dark days.